File: 2021/2/part_1.py
with open("input.txt") as f:
    lines = f.readlines()

h = 0
d = 0

for l in lines:
    if l != "\n":
        a,v = l.split(" ")
        v = int(v)
        if a == "forward":
            h += v
        elif a == "down":
            d += v
        elif a == "up":
            d -= v
        else:
            raise ValueError(l)

print(h*d)

File: 2021/2/part_1.js
const fs = require("fs");

const lines = fs.readFileSync("input.txt", "utf8").trim("\n").replace(/\r/g, "").split("\n");

let h = 0;
let d = 0;

for (l of lines) {
    let [a, v] = l.split(" ");

    v = parseInt(v, 10);

    switch(a) {
        case "forward":
            h += v;
            break;
        case "down":
            d += v;
            break;
        case "up":
            d -= v;
            break;
    }
}

console.log(h * d);
